### DISCUSSION TITLE
defaultValues with field arrays and yup stopped working in 6.x

### DISCUSSION THREAD

**Discussion by mortenk** [react-hook-form/react-hook-form#3280] [Q&A]
Our code stopped working when upgrading to 6.x. 
Input fields are no longer populated with default values. 
Version 5, working:
https://codesandbox.io/s/react-hook-form-array-default-values-572-egv64
Version 6, not working:
https://codesandbox.io/s/react-hook-form-array-default-values-696-jigu1

**Comment by bluebill1049** [comment]
maybe take a look at the changelog first? https://github.com/react-hook-form/react-hook-form/blob/master/CHANGELOG.md

**Reply by mortenk** [reply]
I did, but I couldn't spot anything

**Reply by mortenk** [reply]
I tracked it down, it broke between 6.3.0 (last known good) and 6.3.1 (first version to fail)
There is nothing at all in the changelog for these versions

**Reply by bluebill1049** [reply]
Thanks for the identification, I will update changelog, i am pretty sure this is not an intentional change with `defaultValue`, rather bug which trigger the change, but i will update the doc to reflect that, thanks for your assistant.

**Reply by mortenk** [reply]
This has been working as expected since at least 5.x, now it is broken.
How can it not be a bug?
It breaks in a patch version and you say it's intentional?
Commit 634bb08 is the cause of this breakage

**Reply by bluebill1049** [reply]
This is the best opportunity for you to contribute if you truly believe without defaultValue will not introduce bug with all useFieldArray action. I would recommend you fork the repo, have some deep investigation before making a judgment that I can work without defaultValue. On the other hand, if it truly works and passes all unit tests and integration I am more than happy to review your PR. This is in the doc from the very beginning of V6, you can go watch the video tutorial. It may work (but with a bug) before 6.3, but it's always required to have `defaultValue` to avoid issues while shuffling the array field if may work for your use case, but this is not made only work single use case.

**Reply by mortenk** [reply]
Great idea. I created a PR with a fix and a test for this case.

**Reply by bluebill1049** [reply]
Thanks for the PR, going to merge it with your sold effort, however, I don't think it solves the problem without defaultValue, you will still need to supply a defaultValue. Yes it's passing the test with your unit test, however, it will fail with useFieldArray's action, you can try to remove all the defaultValue in the cypress tests and see if my assumption is valid.
